### PR TITLE
Create and Save Map

### PR DESCRIPTION
**Pull request by ghost**
Demonstrates creating and saving a map to a user's own portal.

### CODE REVIEW

**Comment by ghost** [comment]
@vGandhi and @john0005 could you please review?

File: src/main/java/com/esri/samples/map/create_and_save_map/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<h1>Create and Save Map</h1>
+
+<p>How to create and save a map to your own portal.</p>
+
+<p><img src="CreateAndSaveMap.png"/></p>
+
+<h2>How to use the sample</h2>
+
+Fill in your portal and registered app credentials of the starting dialog to authenticate. Then choose
```
Comment (john0005) [inline src/main/java/com/esri/samples/map/create_and_save_map/README.md]: I like the structure that android has here, https://github.com/Esri/arcgis-runtime-samples-android/tree/master/create-save-map. Explains to the person where this information can be found.

File: src/main/java/com/esri/samples/map/create_and_save_map/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<h1>Create and Save Map</h1>
+
+<p>How to create and save a map to your own portal.</p>
+
+<p><img src="CreateAndSaveMap.png"/></p>
```
Comment (john0005) [inline src/main/java/com/esri/samples/map/create_and_save_map/README.md:5]: Like this better if a gif. Could start like below and end with saving.
login3
login2

Comment (john0005) [reply src/main/java/com/esri/samples/map/create_and_save_map/README.md:5]: When you get to the username/password part does it let you login in? When I try to sign in it says it is signing in but never does.

File: src/main/java/com/esri/samples/map/create_and_save_map/CreateAndSaveMapController.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    String worldElevationService = "http://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer";
+    ArcGISMapImageLayer worldElevation = new ArcGISMapImageLayer(worldElevationService);
+    worldElevation.loadAsync();
+
+    String worldCensusService = "http://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer";
+    ArcGISMapImageLayer worldCensus = new ArcGISMapImageLayer(worldCensusService);
+    worldCensus.loadAsync();
+
+    layersList.getItems().addAll(worldElevation, worldCensus);
+    layersList.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
```
Comment (john0005) [inline src/main/java/com/esri/samples/map/create_and_save_map/CreateAndSaveMapController.java:95]: Lead me to believe that I can show more than one layer, however application only lets me select one at a time.

**Comment by john0005** [comment]
Some initial thoughts. When I can actually get it to login in I will review some more.

**Comment by ghost** [comment]
will address after next week

File: src/main/java/com/esri/samples/map/create_and_save_map/CreateAndSaveMapController.java (hunk 3 above)
Comment (ghost) [reply src/main/java/com/esri/samples/map/create_and_save_map/CreateAndSaveMapController.java:95]: Hold CTRL or SHIFT to select multiple

File: src/main/java/com/esri/samples/map/create_and_save_map/README.md (hunk 1 above)
Comment (ghost) [reply src/main/java/com/esri/samples/map/create_and_save_map/README.md]: Good idea. I'll add that

File: src/main/java/com/esri/samples/map/create_and_save_map/README.md (hunk 2 above)
Comment (ghost) [reply src/main/java/com/esri/samples/map/create_and_save_map/README.md:5]: I've sent you credentials to use. As for the first comment, the sample is really more about saving than oauth, so I'd rather not overly complicate the image.

**Comment by ghost** [comment]
@john0005 This is ready for a second review.

File: src/main/java/com/esri/samples/map/create_and_save_map/AuthenticationDialog.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.esri.arcgisruntime.security.OAuthConfiguration;
+
+/**
+ * Custom dialog for getting an OAuthConfiguration.
+ */
+class AuthenticationDialog extends Dialog<OAuthConfiguration> {
+
+  @FXML
+  private TextField portalURL;
```
Comment (john0005) [inline src/main/java/com/esri/samples/map/create_and_save_map/AuthenticationDialog.java:34]: Which way do we want these to look, some samples I been doing have it all on one line.
`@FXML private TextField portalURL;`

File: src/main/java/com/esri/samples/map/create_and_save_map/README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  <li>Create an <code>ArcGISMap</code> with a <code>Basemap</code> and operational layers</li>
+  <li>Create an <code>OAuthConfiguration</code> with your portal and app credential.</li>
+  <li>Add the configuration to the <code>AuthenticationMangager</code></li>
+  <li>Create a <code>Portal</code> and load it. Use a custom <code>AuthenticationChallengeHandler</code> to 
+  authenticate with your username and password</li>
+  <li>Once authenticated, save the map by calling <code>map.saveMapAsAsync()</code>, passing in the title, tags, 
+  description, and portal folder</li>
+</ol>
+
+<h2>Authentication</h2>
```
Comment (john0005) [inline src/main/java/com/esri/samples/map/create_and_save_map/README.md]: Might also need to say something about adding a folder to their `My Content` section. If they don't have any folders they will not be able to save anything on the application.
content

**Review by john0005 — CHANGES_REQUESTED** [review]

File: src/main/java/com/esri/samples/map/create_and_save_map/README.md (hunk 2 above)
Comment (john0005) [reply src/main/java/com/esri/samples/map/create_and_save_map/README.md:5]: Gif not need just confusing what to do at first without any steps.

**Comment by john0005** [comment]
Looks good. If we could some how add what exactly what the user needs enter to run the application that would be nice.

I know for the portal url you can put `http://www.arcgis.com/` and the Client ID is easy to find out once you make you application. The only hard part is what to put for Redirect URI.

I feel like the sample will be hard to get working if you don't know how to get the Redirect URI working. I know the one that you sent me can work for anyone but not sure for if that can be give out.

**Comment by ghost** [comment]
@john0005 Can you take another look? Also, other sdks put this in the map category so we'll go with that for now.

**Review by john0005 — APPROVED** [review]

**Comment by john0005** [comment]
Looks good. One thing that I noticed that could be added is a Progress Indicator when the save button is hit.

**Comment by ghost** [comment]
@john0005 good call. I've added the progress indicator. Merging.

**Pull request merged**